### PR TITLE
Feature/token paths

### PR DESCRIPTION
**Pull request by Jwilson1172**
# Description

This PR will bring protected paths to all of the endpoints via an `@app.middleware` decorated function that checks the value of the token in the `Authorization` request header.

## Type of change

- [X] New feature (non-breaking change which adds functionality)
- [X] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Change Status

- [X] Complete, tested, ready to review and merge

## Has This Been Tested

- [X] Yes

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] My code has been reviewed by at least one peer
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] There are no merge conflicts

## Additional Comments:

`note` : Breaks Unauthorized user's functionality (intended purpose)

Merging to main will cause any request without "Authorization" header set to the value that is stored in the environment variable DS_SECRET_TOKEN to fail resulting in a 403 status code : Forbidden

### CODE REVIEW

File: project/app/main.py [hunk 1]
```diff
@@ -1,26 +1,63 @@
...
+@app.middleware("http")
+async def check_auth_header(request: Request, next_call):
+    """
+    Function acts as a custom middlewear for http requests, checks if the
+    request header contains Authorization and if the value is equal to the
+    expected value that is stored in memory.
+    Arguments
+    -----------
+    `request`: {fastapi.requests.Request} - requesting client
+    `next_call`: {typing.Callable} - callable function that the middlewear
```
Comment (schase15) [inline project/app/main.py]: * middleware

File: project/app/main.py [hunk 2]
```diff
@@ -1,26 +1,63 @@
...
+    is intercepting
+    Returns
+    ---------
+    `response`: {fastapi.responses.Response} - response from calling the
+    function
+    """
+    bad_response = Response(
+        status_code=403, content="PATH FORBBIDEN", media_type="text/html"
```
Comment (schase15) [inline project/app/main.py]: Typo? Should it be `"PATH FORBIDDEN"`

**Review by schase15 — DISMISSED** [review]
Found a couple small typos

File: project/app/main.py (hunk 2 above)
Comment (Jwilson1172) [reply project/app/main.py]: fixed with commit: 8a0aa51d7fa53455362125f7e54aafca6499821d

File: project/app/main.py (hunk 1 above)
Comment (Jwilson1172) [reply project/app/main.py]: fixed with commit: 8a0aa51d7fa53455362125f7e54aafca6499821d

**Review by schase15 — DISMISSED** [review]

File: project/app/main.py [hunk 3]
```diff
@@ -1,26 +1,63 @@
...
+    description="A RESTful API for the Story Squad Project",
+    version="0.1",
+    docs_url="/",
 )
 
+
+@app.middleware("http")
+async def check_auth_header(request: Request, next_call):
+    """
+    Function acts as a custom middlewear for http requests, checks if the
```
Comment (lorischl-otter) [inline project/app/main.py]: middleware typo

**Review by lorischl-otter — CHANGES_REQUESTED** [review]
Thanks for all the detailed commits and description, Joe! And good catch on the typos, Steven. I unresolved one of them, because there's still a small typo with FORBIDDEN, and then I found one additional 'middleware' typo. Otherwise looks good, and I edited the PR description to include that code has been reviewed. Thanks for getting this implemented!

File: project/app/main.py (hunk 2 above)
Comment (lorischl-otter) [reply project/app/main.py]: Not quite fixed -- missing one pesky 'D' in FORBIDDEN

File: project/app/main.py (hunk 3 above)
Comment (Jwilson1172) [reply project/app/main.py]: fixed with commit:5d1c772772bee984e014c24940ccc5575087c7c0

**Review by schase15 — APPROVED** [review]

**Review by lorischl-otter — APPROVED** [review]

**Pull request merged**